core/src/main/clojure/xtdb/sql/plan.clj

```diff
@@ -155,10 +155,26 @@
     (list 'cast e :f32)
     [:approximate_numeric_type "DOUBLE" "PRECISION"]
     (list 'cast e :f64)
-
+            
+    [:datetime_type "DATE"]
+    (list 'cast e [:date :day])
+    [:datetime_type "TIME"]
+    (list 'cast e [:time-local :micro])
```

Comment:
When casting to temporal, need to specify the specific type of temporal we want. Went with `:<temporal> :micro` in most cases, given that seems to be fairly typically used (same with `:date :day`). Happy to change this - could also use `:nano` here.

Reply:
This seems reasonable for now.